src/core/dev/cq_mgr_tx.cpp

```diff
@@ -204,6 +205,18 @@ cq_mgr_tx *cq_mgr_tx::get_cq_mgr_from_cq_event(struct ibv_comp_channel *p_cq_cha
     return p_cq_mgr;
 }
 
+std::string cq_mgr_tx::wqe_to_hexstring(uint16_t index) const
+{
+    const auto wqe = static_cast<const uint8_t *>(m_hqtx_ptr->m_mlx5_qp.sq.buf) + 64 * index;
+
+    std::ostringstream oss;
+    for (uint8_t i = 0; i < m_hqtx_ptr->m_mlx5_qp.cap.max_inline_data; ++i) {
```

Comment:
Using max_inline_data is not good here. Wqe may vary in size.

Reply:
you have the credits field inside sq_wqe_prop which should specify the number of weqbbs this wqe consumed. Comment there says 'usually' so need to verify.